src/components/TransactionForm/index.js

```diff
@@ -10,33 +10,49 @@ import {
 } from './styled';
 import Button from '../Button';
 
-const TransactionForm = ({item, setReadOnly}) => {
-  const [date, setDate] = useState((item && item.date) || new Date().toISOString().split('T')[0]);
+const TransactionForm = ({item, setItemsToEdit}) => {
+  
+  const [date, setDate] = useState((item && item.date) || Date.now());
   const [title, setTitle] = useState((item && item.title) || '');
   const [amount, setAmount] = useState((item && item.amount) || '');
-  const [type, setType] = useState((item && item.type) || 'income');
+  const [type, setType] = useState((item && item.type) || 'Income');
+  const [category, setCategory] = useState((item && item.category) || 'Salary');
 
-  const { addTransaction, replaceTransaction } = useTransactionContext();
+  const {
+    addTransaction,
+    replaceTransaction,
+    incomeCategories,
+    expenseCategories
+  } = useTransactionContext();
+
+  const categories = type === 'Income' ? incomeCategories : expenseCategories;
+
+  const typeHandler = (type) => {
+    setType(type);
+    setCategory(type === 'Income' ? 'Salary' : 'Food');
+  };
 
   const onSubmit = (evt) => {
     evt.preventDefault();
     const transaction = {
       id: item ? item.id : uuidv4(),
       date,
       type,
+      category,
       title,
       amount,
     };
 
     if (item) {
       replaceTransaction(item.id, transaction);
-      setReadOnly(true);
+      setItemsToEdit(prevState => prevState.filter((oldId) => oldId !== item.id))
     } else {
       addTransaction(transaction);
-      setDate(new Date().toISOString().split('T')[0]);
+      setDate(Date.now());
       setTitle('');
       setAmount('');
-      setType('income');
+      setType('Income');
```

Comment:
UX point: the Type and date are better to stay on the state the user's last set
not reset by you, imagen if I want to add multiple transactions that happened on the same day